app/stac_api/collection_summaries.py

```diff
@@ -0,0 +1,203 @@
+            collection.save()
+
+    except KeyError as err:
+        logger.error(
+            "Error when updating collection's summaries values due to asset deletion: %s", err
+        )
+        raise ValidationError(_(
+            "Error when updating collection's summaries values due to asset deletion."
+        ))
+
+
+def update_summaries_on_asset_insert(collection, asset):
+    '''
+    updates the collection's summaries on an asset's insertion or raises
+    errors when this fails.
+    :param collection: collection, for which the summaries probably need an update
+    :param asset: asset thats being inserted
+    For all the given parameters this function checks, if the corresponding
+    parameters of the collection need to be updated. If so, they will be either
+    updated or an error will be raised, if updating fails.
+    '''
+
+    try:
+        if asset.geoadmin_variant and \
+            asset.geoadmin_variant not in collection.summaries["geoadmin:variant"]:
+            collection.summaries["geoadmin:variant"].append(asset.geoadmin_variant)
+            collection.summaries["geoadmin:variant"].sort()
+            collection.save()
+
+        if asset.proj_epsg and \
+                asset.proj_epsg not in collection.summaries["proj:epsg"]:
+            collection.summaries["proj:epsg"].append(asset.proj_epsg)
+            collection.summaries["proj:epsg"].sort()
+            collection.save()
+
+        if asset.eo_gsd and not float_in(asset.eo_gsd, collection.summaries["eo:gsd"]):
+            collection.summaries["eo:gsd"].append(asset.eo_gsd)
+            collection.summaries["eo:gsd"].sort()
+            collection.save()
+
+    except KeyError as err:
+        logger.error(
+            "Error when updating collection's summaries values due to asset insert: %s", err
+        )
+        raise ValidationError(_(
+            "Error when updating collection's summaries values due to asset insert."
+        ))
+
+
+def update_summaries_on_asset_update(collection, asset, old_values):
+    '''
+    updates the collection's summaries on an asset's update or raises
+    errors when this fails.
+    :param collection: collection, for which the summaries probably need an update
+    :param asset: asset thats being updated
+    :param old_values: (optional) list with the original values of asset's
+    eo_gsd, geoadmin_variant and proj_epsg.
+    For all the given parameters this function checks, if the corresponding
+    parameters of the collection need to be updated. If so, they will be either
+    updated or an error will be raised, if updating fails.
+    '''
+
+    original_eo_gsd = old_values[0]
+    original_geoadmin_variant = old_values[1]
+    original_proj_epsg = old_values[2]
+
+    asset_qs = None
+
+    try:
+
+        if original_geoadmin_variant != asset.geoadmin_variant:
+
+            if asset.geoadmin_variant and \
+            asset.geoadmin_variant not in collection.summaries["geoadmin:variant"]:
```

Comment:
Indentation issue. Might be a bug of `yapf` though. Anyway breaking long if statement can also be done with `()` instead of `/` different style both are acceptable. I've noticed that `yapf` tends to support `()` better than `\`:
e.g. : 
```
            if (asset.geoadmin_variant 
                and asset.geoadmin_variant not in collection.summaries["geoadmin:variant"]):
```

Reply:
now using `()` instead of `\`